src/json_serialization.hpp

```diff
@@ -24,6 +24,80 @@ namespace madrona_gpudrive
         return 0.0f; // Default z value if not provided
     }
 
+    void from_json(const nlohmann::json &j, TrafficLightState &tl_state)
+    {
+        // Get the lane ID from the key and store it
+        tl_state.laneId = std::stoi(j.begin().key());
+
+        // Set number of states to the size of the state array
+        size_t numStates = std::min(j.at("state").size(), static_cast<size_t>(consts::kTrajectoryLength));
+        tl_state.numStates = numStates;
+
+        // Process each timestep
+        for (size_t t = 0; t < numStates; t++) {
+            // Get the state string and convert to enum
+            if (t < j.at("state").size()) {
+                std::string state_str = j.at("state")[t];
+                if (state_str == "unknown") {
+                    tl_state.state[t] = TLState::Unknown;
+                } else if (state_str == "stop") {
+                    tl_state.state[t] = TLState::Stop;
+                } else if (state_str == "caution") {
+                    tl_state.state[t] = TLState::Caution;
+                } else if (state_str == "go") {
+                    tl_state.state[t] = TLState::Go;
+                } else {
+                    tl_state.state[t] = TLState::Unknown;
+                }
+            } else {
+                tl_state.state[t] = TLState::Unknown;
+            }
+
+            // Get coordinates
+            if (t < j.at("x").size()) {
+                tl_state.x[t] = j.at("x")[t];
+            } else {
+                tl_state.x[t] = 0.0f;
+            }
+
+            if (t < j.at("y").size()) {
+                tl_state.y[t] = j.at("y")[t];
+            } else {
+                tl_state.y[t] = 0.0f;
+            }
+
+            if (t < j.at("z").size()) {
+                tl_state.z[t] = j.at("z")[t];
+            } else {
+                tl_state.z[t] = 0.0f;
+            }
+
+            // Get time index and lane id
+            if (t < j.at("time_index").size()) {
+                tl_state.timeIndex[t] = j.at("time_index")[t];
+            } else {
+                tl_state.timeIndex[t] = -1;
+            }
+
+            if (t < j.at("lane_id").size()) {
+                // Lane ID is already stored from key, but also save per timestep
+                tl_state.laneIds[t] = j.at("lane_id")[t];
+            } else {
+                tl_state.laneIds[t] = tl_state.laneId;
+            }
+        }
+
+        // Fill any remaining timesteps with default values
+        for (size_t t = numStates; t < consts::kTrajectoryLength; t++) {
+            tl_state.state[t] = TLState::Unknown;
+            tl_state.x[t] = 0.0f;
+            tl_state.y[t] = 0.0f;
+            tl_state.z[t] = 0.0f;
```

Comment:
we will have different default values between the invalid objects (set in the conversion script to -1000) and the invalid TL states.  I think it makes sense to use the same default value for the two